### PR TITLE
Draft usage of pixi task for notebook analysis.

### PR DESCRIPTION
**Pull request by kklein**
This PR should make it a little more convenient to reproduce the behaviour of the `mypy-example-nbs` CI job locally.

# Checklist

- [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

File: pixi.toml [hunk 1]
```diff
@@ -6,6 +6,9 @@ platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
 [tasks]
 postinstall = "pip install --no-build-isolation --no-deps --disable-pip-version-check -e ."
 coverage = "pytest tests --cov=metalearners --cov-report=xml --cov-report term-missing --color=yes"
+# Note that this task will fail on systems without bash.
+nb-convert = "nb-convert.sh"
```
Comment (pavelzw) [inline pixi.toml:10]: this probably won't work on windows unless cygwin/msys2 is installed but i guess this is fine

**Review by pavelzw — APPROVED** [review]
for easier reproducible notebooks i suggest taking a look at https://marimo.io/ 😉 it's a neat project

File: pixi.toml (hunk 1 above)
Comment (kklein) [reply pixi.toml:10]: Yup, that's what I was trying to allude to with the commend above... :/

**Comment by kklein** [comment]
> for easier reproducible notebooks i suggest taking a look at https://marimo.io/ 😉 it's a neat project

Thanks for the pointer! Do you happen to know whether it intregrates with mkdocs/sphynx, too?

Our main reason for using jupyter notebooks right now is to be able to generate code-heavy examples for our docs.

File: pixi.toml (hunk 1 above)
Comment (pavelzw) [reply pixi.toml:10]: oh i overlooked this comment 😅

**Comment by pavelzw** [comment]
> Do you happen to know whether it integrates with mkdocs/sphynx, too?

**Comment by pavelzw** [comment]
> Do you happen to know whether it integrates with mkdocs/sphynx, too?

There are marimo islands ([docs](https://docs.marimo.io/guides/island_example.html)) that can generate HTML that can be easily embedded.
Otherwise you could use their web app to create an embed (see https://github.com/pyodide/pyodide-blog/blob/34de33029ee9246e507b1f9621f9239cea27ee8f/content/posts/marimo.md?plain=1#L72) but this is a bit clunky if your example code is in your repo.

But I don't think that the tooling is there yet to autogenerate mkdocs or sphinx content.
If I find some time in the future, looking into a mkdocs plugin is on my todo list.

xref https://discord.com/channels/1059888774789730424/1214483174361534474/1214561254035951636, https://discord.com/channels/1059888774789730424/1227738354175578203/1262785309201137734

**Pull request merged**